src/soc/amd/rome/src/lib.rs

```diff
@@ -2,44 +2,46 @@
 #![feature(llvm_asm)]
 
 pub mod hsmp;
-
 use hsmp::HSMP;
 
-pub struct SOC {}
+pub struct SOC {
+    hsmp: HSMP,
+    topology: df::FabricTopology,
+}
 
 impl SOC {
     pub fn new() -> SOC {
-        Self {}
+        let hsmp = HSMP::new();
+        let topology = df::FabricTopology::new();
+        Self { hsmp, topology }
```

Comment:
Maybe it's not good to make it copy those

Reply:
so, looks like it was not copying it: I don't think FabricTopology has the Copy trait. I *think* the compiler did what is should. I changed it anyway.